hazelcast/src/main/java/com/hazelcast/internal/partition/impl/PartitionReplicaStateChecker.java

```diff
@@ -162,7 +161,7 @@ private boolean hasMissingReplicaOwners() {
                 }
 
                 if (clusterService.getMember(address) == null
-                        && (!isClusterNotActive || !clusterService.isMemberRemovedWhileClusterIsNotActive(address))) {
+                        && (clusterState.isJoinAllowed() || !clusterService.isMemberRemovedWhileClusterIsNotActive(address))) {
```

Comment:
Actually checking `IN_TRANSITION` is redundant here, because to be able to change cluster state, we ensure that there are no ongoing/pending migrations.